docs/Actions/inject-data.md

```diff
@@ -0,0 +1,274 @@
+---
+id: inject-data
+title: Inject Data
+sidebar_label: Inject Data
+---
+
+> Audience: [`Business Users`](/docs/audience#business-users)<br/>
+> Skill Prerequisites: `None`
+
+An action that lets you add data as tokens to the current execution context. Inject data can be used to load data that other actions other actions down the stack need.
+
+## `Typical Use Cases`
+
+- Generate a new token
+- Concatenate 2 tokens
+- Overwrite a token
+- Initialize variables with empty strings
+
+## `Don't use it to`
+
+- Execute a Variables token - use [Apply Tokens](/docs/Actions/apply-tokens.md) or [Execute Tokens](/docs/Actions/execute-tokens.md) actions instead
+- Apply tokens inside a text from different sources - use [Apply Tokens](/docs/Actions/apply-tokens.md) or [Execute Tokens](/docs/Actions/execute-tokens.md) actions instead
+- Create a token that is used in another execution context - store the data in a [hidden field](/docs/Actions/hidden-field.md), cookies or server session instead
+
+## `Related Actions`
+
+| Action Name | Description|
+|-------------|------------|
+| [Run SQL Query](/docs/Actions/run-sql-query.md) | Executes an SQL statement and captures the output, so that it can be used in the form using Inject Data.|
+
+## `Input Parameter Reference`
+
+| Parameter     | Description                           | Supports Variables |
+|---------------|---------------------------------------|--------------------|
+| Name/Value | The first parameter refers to the name of the new token that will be created. The second one refers to the value assigned. The action uses a key-value (name-value) concept. | Yes |
+
+## `Output Parameters Reference`
+
+| Parameter | Description |
+|-----------|-------------|
+| Name | The Name parameter is stored as a token for further use anywhere in the form. |
```

Comment:
I don't see an Output Parameter named "Name" in the action. If you refer to the Name/Value collection, I suggest that we move the entire parameter under Output Parameters section and we don't cover it in Input at all.

Reply:
I saw on Radu's commit another structure(that seemed good), so i used it here. I propose we use the same structure everywhere we have a key-value pair.